app/models/post.rb

```diff
@@ -305,8 +305,11 @@ def cook(raw, opts = {})
     options = opts.dup
     options[:cook_method] = cook_method
 
-    post_user = self.user
-    options[:user_id] = post_user.id if post_user
+    # This user_id is used to fetch a user which is populated into the
+    # currentUser option in our markdown pipeline. The last editor is used
+    # to ensure that any markdown rules using the current_user for a guardian
+    # can have the correct permission controls.
```

Comment:
Honestly even after multiple reading of this paragraph, Im not sure I understand it. Can you think of a more precise example of what is the reason of this line, and why you felt it had to be explained please.

Reply:
Basically, the markdown flow may have guardian checks that require a user. We are changing this to be the last editing user rather than the creating user so we use the permissions of who last touched the post/message, since it's more up to date. I have an internal topic about this if you want to see more. But I will update this description to be simpler + with an example.